appinventor/components/src/com/google/appinventor/components/runtime/Canvas.java

```diff
@@ -1264,6 +1266,96 @@ public void DrawLine(int x1, int y1, int x2, int y2) {
     view.invalidate();
   }
 
+  /**
+   * Draws a shape on the canvas, 
+   * point list should be a list contains sub-lists with two number which represents a coordinate
+   *
+   * @param pointList  a list of points, should contains sub-lists with two number which represents a coordinate
+   * @param fill  true for filled shape; false for shape outline
+   */
+  @SimpleFunction
+  public void DrawShape(YailList pointList, boolean fill) {
+    Path path = parsePath(parsePointList(pointList));
+    if (path == null) {
```

Comment:
Rather than returning null from these functions, consider throwing `IllegalArgumentException` and then dispatching the error in a `catch` block.